### PR TITLE
Feature/add architecture to feature

### PR DESCRIPTION
**Pull request by AbdulWahab3181**
## Describe your changes

https://www.loom.com/share/b8a6b263073c4434b56c8e16af459518

## Issue ticket number and link

Closes https://github.com/stakwork/sphinx-tribes-frontend/issues/515

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -51,6 +52,9 @@ const modeDispatchPages: Record<AppMode, () => React.ReactElement> = {
           <Route path="/leaderboard">
             <LeaderboardPage />
           </Route>
```
Comment (elraphty) [inline src/pages/index.tsx:54]: @AbdulWahab3181 please pull from master there is a route and component for this already ```sh          
<Route path="/feature/:feature_uuid">
     <WorkspaceFeature />
 </Route>
```

File: src/store/main.ts [hunk 2]
```diff
@@ -2294,6 +2317,51 @@ export class MainStore {
     }
   }
 
+  async getUserFeatureByUuid(uuid: string): Promise<Feature | undefined> {
+    try {
+      if (!uiStore.meInfo) return undefined;
+      const info = uiStore.meInfo;
+      console.log('TribesURL', TribesURL);
+      const r: any = await fetch(`${TribesURL}/features/${uuid}`, {
+        method: 'GET',
```
Comment (elraphty) [inline src/store/main.ts:2326]: Pull from master, some of these functions are already implemented

Comment (AbdulWahab3181) [reply src/store/main.ts]: @elraphty Addressed

**Comment by AbdulWahab3181** [comment]
**Demo**

https://www.loom.com/share/39bd103e441942fbb7fe5fca0dee0bd2

**Comment by elraphty** [comment]
> **Demo**
> 
> https://www.loom.com/share/39bd103e441942fbb7fe5fca0dee0bd2

Okay, @AbdulWahab3181 DId you merge from master or rebased?

**Comment by AbdulWahab3181** [comment]
> > **Demo**
> > https://www.loom.com/share/39bd103e441942fbb7fe5fca0dee0bd2
> 
> Okay, @AbdulWahab3181 DId you merge from master or rebased?

I did rebased from master

**Comment by elraphty** [comment]
> > > **Demo**
> > > https://www.loom.com/share/39bd103e441942fbb7fe5fca0dee0bd2
> > 
> > 
> > Okay, @AbdulWahab3181 DId you merge from master or rebased?
> 
> @elraphty I did rebased from master

Okay, @AbdulWahab3181 tested it works fine, add a cypress test to it so I can merged

**Pull request merged**